### ISSUE TITLE
Growatt : 'Export to grid' information missing ?

### ISSUE THREAD

**Issue by MiosDK** [biemond/solaredge.modbus#78]
Hi, 
I have a Growatt 3600 hybrid inverter where I need the information regarding: 'export to grid' is it possible to this information? 
![Screenshot_20240521_194837_ShinePhone~3](https://github.com/biemond/solaredge.modbus/assets/170458604/169c2744-cb1f-4cce-b018-1ee22b92759b)

**Comment by biemond** [comment]
hey, what does this say on yours

```
    "measure_power.gridoutput": {
      "decimals": 0,
      "title": {
        "en": "Grid output",
        "de": "Netzexport"
      }
    },
```

        "gridOutputPower": [40, 2, 'UINT32', "Grid Output Power", -1],

**Comment by biemond** [comment]
or do you mean total of today export.  

can you find out what register I need to use for that.?

**Comment by MiosDK** [comment]
> hey, what does this say on yours
> 
> ```
>     "measure_power.gridoutput": {
>       "decimals": 0,
>       "title": {
>         "en": "Grid output",
>         "de": "Netzexport"
>       }
>     },
> ```
> 
> ```
>     "gridOutputPower": [40, 2, 'UINT32', "Grid Output Power", -1],
> ```

I dont know to see that ?

**Comment by MiosDK** [comment]
> or do you mean total of today export.
> 
> can you find out what register I need to use for that.?

I need the actual export 'up front' not all Day, I should use it regarding, do you know what I Mean? I dont have a register overview, do you know where I can find the register ?

**Comment by biemond** [comment]
you should have grid output on your device , what device did you choose?

<img width="1029" alt="Screen Shot 2024-05-21 at 21 44 08" src="https://github.com/biemond/solaredge.modbus/assets/1085004/f3bcc085-2ec7-461e-a331-4a54948fc6ee">

**Comment by MiosDK** [comment]
> you should have grid output on your device , what device did you choose?
> 
> <img alt="Screen Shot 2024-05-21 at 21 44 08" width="1029" src="https://private-user-images.githubusercontent.com/1085004/332561138-f3bcc085-2ec7-461e-a331-4a54948fc6ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzMjExMzYsIm5iZiI6MTcxNjMyMDgzNiwicGF0aCI6Ii8xMDg1MDA0LzMzMjU2MTEzOC1mM2JjYzA4NS0yZWM3LTQ2MWUtYTMzMS00YTU0OTQ4ZmM2ZWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MjFUMTk0NzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGVkNzk3YTYzNTJiOWU0MDk4ZGQ0NjQ1ZTA1ZGU5MjFkNzIxM2E2YjA0ZWMwOTQ2MGZjNmZiMjVhZjE4NTE1OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qppk1DR-cL6Q3AFNSFvsNazGqHk_iL6ulWU1sth_yZI">

This one at the picture, is this the right one ? 

![Screenshot_20240521_214929_Homey~2](https://github.com/biemond/solaredge.modbus/assets/170458604/c8051dda-65e9-4b2a-b6b6-92607d6b6dd4)

**Comment by MiosDK** [comment]
> > you should have grid output on your device , what device did you choose?
> > <img alt="Screen Shot 2024-05-21 at 21 44 08" width="1029" src="https://private-user-images.githubusercontent.com/1085004/332561138-f3bcc085-2ec7-461e-a331-4a54948fc6ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzMjExMzYsIm5iZiI6MTcxNjMyMDgzNiwicGF0aCI6Ii8xMDg1MDA0LzMzMjU2MTEzOC1mM2JjYzA4NS0yZWM3LTQ2MWUtYTMzMS00YTU0OTQ4ZmM2ZWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MjFUMTk0NzE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGVkNzk3YTYzNTJiOWU0MDk4ZGQ0NjQ1ZTA1ZGU5MjFkNzIxM2E2YjA0ZWMwOTQ2MGZjNmZiMjVhZjE4NTE1OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qppk1DR-cL6Q3AFNSFvsNazGqHk_iL6ulWU1sth_yZI">
> 
> This one at the picture, is this the right one ?
> 
> ![Screenshot_20240521_214929_Homey~2](https://private-user-images.githubusercontent.com/170458604/332562815-c8051dda-65e9-4b2a-b6b6-92607d6b6dd4.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzMjEzNjEsIm5iZiI6MTcxNjMyMTA2MSwicGF0aCI6Ii8xNzA0NTg2MDQvMzMyNTYyODE1LWM4MDUxZGRhLTY1ZTktNGIyYS1iNmI2LTkyNjA3ZDZiNmRkNC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyMVQxOTUxMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYTg4NWQ3MGZjZDI0ZTg3MDZhZWY1NjA1NjJhMGNkMzAzYjQwYjJiZDRiZWU3MDRlZTg0N2ViNzRmZTZhMjc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ecJN3_slElidJDvHinANtmVOSfXTvXItsk13u52-dmE)

I need the actual 'export to grid' readings, so when the battery is 100% and the sun panels is produceing ex. 1200watt and the house is using 600watt the last 600 watt is to be exported to grid, therefor I will start something in the house that will use the the power insted.

**Comment by biemond** [comment]
I think you can use the TL one.  and you send me a diagnostic report.  but grid output is the 1

**Comment by MiosDK** [comment]
> I think you can use the TL one. and you send me a diagnostic report. but grid output is the 1

TL is also ok can I see, but there is not more options to this matter, I have tried grid output, its not the one to use in this matter, the name should be 'export to grid' can upload a modbus PDF maybe I can find the correct register ?

**Comment by MiosDK** [comment]
> > I think you can use the TL one. and you send me a diagnostic report. but grid output is the 1
> 
> TL is also ok can I see, but there is not more options to this matter, I have tried grid output, its not the one to use in this matter, the name should be 'export to grid' can upload a modbus PDF maybe I can find the correct register ? I can see there is a 'Grid export YIELD' thats the total for the Day, I need this information when it change immediately, do you know what meen ? 😃

**Comment by biemond** [comment]
when it is not there , I need to calculate it or find the right register.   but you can also do this yourself with solar/batt output minus the load. 

what is not ok on grid output?

**Comment by MiosDK** [comment]
> when it is not there , I need to calculate it or find the right register. but you can also do this yourself with solar/batt output minus the load.
> 
> what is not ok on grid output?

I got some information that the modbus registre is 122 + 123 can you fix it ?

**Comment by biemond** [comment]
will check those.

**Comment by MiosDK** [comment]
> will check those.

New picture from the register here : 
![FB_IMG_1716368424589](https://github.com/biemond/solaredge.modbus/assets/170458604/5ee47fed-ad2a-4d09-bce7-4c32209a6a20)

**Comment by biemond** [comment]
looks like it I have already have it on the device or as flow actions , else I will enable it https://github.com/biemond/solaredge.modbus/blob/main/drivers/growattwithbatt_tl/device.ts#L80

https://github.com/biemond/solaredge.modbus/blob/main/drivers/growattwithbatt_tl/device.ts#L28

**Comment by MiosDK** [comment]
> oh you want a flow trigger action and block total export.
> 
> currently I support these periods with grid, load or battery priority https://github.com/biemond/solaredge.modbus/blob/main/drivers/growattwithbatt_tl/driver.flow.compose.json

I dont want to Block export but if I let ex. 500watt in export I will start a electric healing element to my heat exchanger to use the export watt...

**Comment by MiosDK** [comment]
> looks like it I have already have it on the device or as flow actions , else I will enable it https://github.com/biemond/solaredge.modbus/blob/main/drivers/growattwithbatt_tl/device.ts#L80
> 
> https://github.com/biemond/solaredge.modbus/blob/main/drivers/growattwithbatt_tl/device.ts#L28

Is it the same as the picture ? 
![Uploading FB_IMG_1716368424589.jpg…]()

**Comment by MiosDK** [comment]
I dont think the function Is in the flowcard

**Comment by biemond** [comment]
you are asking with this 122,123 to block export.   

current grid output should be ok for that.  can you upload the pic on github, not replying on the email.  
also send me a diagnostic report.

**Comment by MiosDK** [comment]
> you are asking with this 122,123 to block export.
> 
> current grid output should be ok for that. can you upload the pic on github, not replying on the email. also send me a diagnostic report.

The "only" thing I want, is to see the output mesuring of the 'export to grid' so when its in +xxx watt to grid I want to start my electric device to use the xxx watt in the house and when the export of xxx watt is lens ex. 200watt I want to stop my device to use energy in the house.

I dont know how to send a diagnostic report?

**Comment by MiosDK** [comment]
> I got the developer version but like this on my app
> 
> <img alt="Screenshot 2024-05-22 at 14 56 50" width="1752" src="https://private-user-images.githubusercontent.com/1085004/332804543-b67ff8d2-67ac-4f46-b98b-5d9fd72d6cec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzOTAyOTksIm5iZiI6MTcxNjM4OTk5OSwicGF0aCI6Ii8xMDg1MDA0LzMzMjgwNDU0My1iNjdmZjhkMi02N2FjLTRmNDYtYjk4Yi01ZDlmZDcyZDZjZWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MjJUMTQ1OTU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmMwYTY5ZDVmZjYzYzY1NDY0ZTUwODJjY2VkZjMzODI1MmQxZTlkZDI4NzhjZDlhMjQ1YzkwZjYwNGU0MDA3YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.pizoCmTjq2EG6KIrYzeFLFgAMS0gmjS_uwixOm4EWd8">

SMPL ?

**Comment by biemond** [comment]
yeah select my app and press Create Diagonistic report button

**Comment by MiosDK** [comment]
> yeah select my app and press Create Diagonistic report button

Ahh Done, I have called it MiosDK-2024

**Comment by MiosDK** [comment]
Do you think its possible eith this as I wrote before ? I think the modbus register I wrote before isnt the right one ?



"The "only" thing I want, is to see the output mesuring of the 'export to grid' so when its in +xxx watt to grid I want to start my electric device to use the xxx watt in the house and when the export of xxx watt is lens ex. 200watt I want to stop my device to use energy in the house."

**Comment by MiosDK** [comment]
I have this Programmed in the the Growatt inverter - but I want this in Homey instead - Maybe it can help you in this matter and you can find it in the Modbus registre :

![billede](https://github.com/biemond/solaredge.modbus/assets/170458604/503067a5-7da6-4e0d-9753-83496da9744c)

**Comment by MiosDK** [comment]
I have this Programmed in the the Growatt inverter - but I want this in Homey instead - Maybe it can help you in this matter and you can find it in the Modbus registre :

![billede](https://github.com/biemond/solaredge.modbus/assets/170458604/054537c7-a4e9-48ba-b111-44756582c80d)

**Comment by biemond** [comment]
got your report

**Comment by biemond** [comment]
```
l1_current 99 -1 L1 Current
l2_current 0 -1 L2 Current
l3_current 0 -1 L3 Current
temperature 472 -1 Temperature
status 5 0 Status
inputPower 24300 -1 Input Power
outputPower 23099 -1 Output Power
pv1Voltage 2641 -1 pv1 Voltage
pv2Voltage 1811 -1 pv2 Voltage
gridFrequency 4999 -2 Grid Frequency
gridVoltage 2409 -1 Grid Voltage
gridOutputCurrent 98 -1 Grid Output Current
gridOutputPower 23252 -1 Grid Output Power
todayEnergy 224 -1 Today Energy
totalEnergy 111805 -1 Total Energy
pv1InputPower 23263 -1 pv1 Power
pv2InputPower 1109 -1 pv2 Power
pv1TodayEnergy 162 -1 pv2 Today Energy
pv1TotalEnergy 71547 -1 pv2 Total Energy
pv2TodayEnergy 111 -1 pv2 Today Energy
pv2TotalEnergy 39290 -1 pv2 Total Energy
pvEnergyTotal 111063 -1 pv Total Energy
error 0 0 Error
battDischarge 0 -1 battery Discharge
battCharge 0 -1 battery Charge
battvoltage 544 -1 battery Voltage
battsoc 100 0 battery soc
batttemperature 25 -1 battery Temperature
bmssoc 100 0 bms soc
bmstemperature 25 -1 bms Temperature
bmscyclecount 571 0 bms cycle count
bmshealth 94 0 bms soh
bmsstatus 289 0 bms status
bmserror 0 0 bms error
totalhouseload 10000 -1 Total house Load
priority 0 0 priority
today_grid_import 25 -1 Today's Grid Import
total_grid_import 117827 -1 Total Grid Import
today_grid_export 127 -1 Today's Grid Export
total_grid_export 33160 -1 Total Grid Export
today_battery_output_energy 11 -1 Today's Battery Output Energy
total_battery_output_energy 32254 -1 Total Battery Output Energy
today_battery_input_energy 43 -1 Today's Battery Input Energy
total_battery_intput_energy 28064 -1 Total Battery Input Energy
today_load 140 -1 Today's Load
total_load 199920 -1 Total Load
exportlimitenabled 0 0 Export Limit enable
exportlimitpowerrate 0 -1 Export Limit Power Rate
prioritychange 0 0 Priority
gridfirststopsoc 10 0 GridFirst stop SOC
batfirststopsoc 95 0 BatFirst stop SOC
acchargeswitch 0 0 Batt AC charge switch
gridfirststarttime1 4352 0 Grid First Start Time
gridfirststoptime1 5376 0 Grid First Stop Time
gridfirststopswitch1 0 0 Grid First Stop Switch 1
battfirststarttime1 0 0 Battery First Start Time
battfirststoptime1 0 0 Battery First Stop Time
battfirststopswitch1 1 0 Battery First Stop Switch 1
loadfirststarttime1 0 0 Load First Start Time
loadfirststoptime1 0 0 Load First Stop Time
loadfirststopswitch1 0 0 Load First Stop Switch 1
gridfirst1 from: 17:00 ~ 21:00 state: 0
battfirst1 from: 00:00 ~ 00:00 state: 1
```

**Comment by biemond** [comment]
so this is your power

your export is gridOutputPower 2325 W or not.
solar = 2430 

```
inputPower 24300 -1 Input Power
outputPower 23099 -1 Output Power
gridOutputPower 23252 -1 Grid Output Power
pv1InputPower 23263 -1 pv1 Power
pv2InputPower 1109 -1 pv2 Power
battDischarge 0 -1 battery Discharge
battCharge 0 -1 battery Charge
```

**Comment by MiosDK** [comment]
> so this is your power
> 
> your export is gridOutputPower 2325 W or not. solar = 2430
> 
> ```
> inputPower 24300 -1 Input Power
> outputPower 23099 -1 Output Power
> gridOutputPower 23252 -1 Grid Output Power
> pv1InputPower 23263 -1 pv1 Power
> pv2InputPower 1109 -1 pv2 Power
> battDischarge 0 -1 battery Discharge
> battCharge 0 -1 battery Charge
> ```
New report send - MiosDK-2024 - 2 - a19b31e1-2393-48d2-b3e9-8e3a4a4b4c54

![billede](https://github.com/biemond/solaredge.modbus/assets/170458604/26219330-9065-4ad5-b526-3baad6ff01b2)

**Comment by biemond** [comment]
so you can use grid power, but always do 5 min average else it will be on / off every few seconds

**Comment by MiosDK** [comment]
> so you can use grid power, but always do 5 min average else it will be on / off every few seconds

I dont think that Gridpower is the solution for this, right now I do not export anything because the sun isnt there any more, I have not at view hvor there is 0 KW - right now GRID output is 931W that is what the house is using (the same as Houseload ) and have nothing with export to Grid - is it possible there is another modbus protocol that can be used ?

**Comment by biemond** [comment]
gridOutputPower  should be and it is working for others , not grid power, that is now your batt discharge

**Comment by MiosDK** [comment]
> gridOutputPower

I can not see gridOutputPower  anywhere in the app - I have tried to find gridOutputPower  text in both Growatt ?

**Comment by biemond** [comment]
it is this 1 ```    "measure_power.gridoutput": {
      "decimals": 0,
      "title": {
        "en": "Grid output",
        "de": "Netzexport"
      }
    },```

**Comment by MiosDK** [comment]
> it is this 1 ` "measure_power.gridoutput": { "decimals": 0, "title": { "en": "Grid output", "de": "Netzexport" } },`

Ok, but right now I do not export anything but using power from grid (see pic) and that is what Grid output is showing and not export from sun to grid ?
![Screenshot_20240523_055331_Homey](https://github.com/biemond/solaredge.modbus/assets/170458604/abdc0683-5007-4a39-a535-899885e2a60b)

**Comment by biemond** [comment]
yeah, I see it.  more as load.  It works on someone else as expected. looks your meters are not installed correctly , let check on someone else also.

**Comment by biemond** [comment]
<img width="748" alt="Screenshot 2024-05-23 at 09 49 56" src="https://github.com/biemond/solaredge.modbus/assets/1085004/80ee696b-e21e-4d44-a6e8-955d90a757e8">

**Comment by MiosDK** [comment]
Is it possible there is a difference in witch country the Growatt os placed ? I have the SPH3600 hybrid inverter and a summation meter from electric vendor + smartmeter from Growatt. Do you know if there is others from Denmark there have this solution ? As you can see on this picture, it seems that the import is 0,5 kw, in Homey I think 0,5 kw= bat power + import from grid. Maybe the summation meter is doing something in this matter ? 
The weird thing in this matter that I can do a programming in Growatt app. To do something when sun is exporting to grid with XX watt and shopping again depent how lens Watt is, I think the information is the same in the modbus ? 

![Screenshot_20240523_212029_One UI Home](https://github.com/biemond/solaredge.modbus/assets/170458604/7667daee-8bad-4ddc-9de0-a7a8294dad97)

**Comment by biemond** [comment]
solar power is pv - batt charge or batt discharge

batt discharge is your solar/batt . 
grid output is + for I think it is import.   
load is both combined here

can you check insights if grid output goes negative with a load of sun and a full batt.   I think negative is export.

**Comment by MiosDK** [comment]
What about this mesurering on L1 / Grid output current is this in watt or kw, is it possible that this one is mesuring the output to Grid ? 
![Screenshot_20240523_214523_Samsung Internet](https://github.com/biemond/solaredge.modbus/assets/170458604/1b997dd1-c8ec-49a3-88c5-760cfed00ce0)

**Comment by biemond** [comment]
can you show me your insights graphs for the last 7 days  about grid output. I think sometimes it goes negative . which means export, pos = import

**Comment by MiosDK** [comment]
> solar power is pv - batt charge or batt discharge
> 
> batt discharge is your solar/batt . grid output is + for I think it is import. load is both combined here
> 
> can you check insights if grid output goes negative with a load of sun and a full batt. I think negative is export.

The grid is not going negative as you can see at the pic

![Screenshot_20240523_220733_Homey](https://github.com/biemond/solaredge.modbus/assets/170458604/aef4ead4-1b7b-406a-8f1e-e05e927de56d)

**Comment by biemond** [comment]
no clue, I just pass it on , let me vpn in to some growatt and see if I see the same.

**Comment by MiosDK** [comment]
> no clue, I just pass it on , let me vpn in to some growatt and see if I see the same.

Ok perfect, let me know if I can do.something more in this matter 👍

**Comment by biemond** [comment]
you are on the latest firmware or not?

**Comment by MiosDK** [comment]
> you are on the latest firmware or not?

I dont now the latest firmware number is ? Maybe you know ? And how can I update if my is a old version ? 

![Screenshot_20240523_230218_ShinePhone](https://github.com/biemond/solaredge.modbus/assets/170458604/2de0b90a-2357-4e6a-a126-210cf8eec206)

**Comment by biemond** [comment]
oh you have a sph , make sure you don't use tl device but the normal growatt batt device.

**Comment by MiosDK** [comment]
> oh you have a sph , make sure you don't use tl device but the normal growatt batt device.

Im using this (see pic) what about the firmware is this the newest ? 

![Screenshot_20240521_214929_Homey~2](https://github.com/biemond/solaredge.modbus/assets/170458604/61b557ba-1aba-4bdc-8f01-50a566a5c5c0)

**Comment by MiosDK** [comment]
What about my firmware, is this the newest one ?

**Comment by biemond** [comment]
no clue.

**Comment by biemond** [comment]
got some for sph, let me add these and then you can send me diagnostic report

<img width="1006" alt="Screenshot 2024-05-24 at 21 28 40" src="https://github.com/biemond/solaredge.modbus/assets/1085004/15d8d239-cb26-45d3-8587-04a3c2910ae9">

**Comment by biemond** [comment]
can you test this https://homey.app/a/solaredge.modbus/test/

and send me a report after 5 min or so. like to see the data.

**Comment by MiosDK** [comment]
> can you test this https://homey.app/a/solaredge.modbus/test/
> 
> and send me a report after 5 min or so. like to see the data.

Do I need to uninstall the other version first ?

**Comment by biemond** [comment]
no
just install

**Comment by MiosDK** [comment]
> no just install


New report sent : d81c15ef-bd0b-4a67-8a46-9aef2ad80ff0

**Comment by biemond** [comment]
```
inputPower 0 -1 Input Power
outputPower 9000 -1 Output Power
gridOutputPower 9059 -1 Grid Output Power
pv1InputPower 11 -1 pv1 Power
pv2InputPower 7 -1 pv2 Power
battDischarge 9900 -1 battery Discharge
battCharge 0 -1 battery Charge
totalhouseload 9900 -1 Total house Load
pactouserr 0 -1 import from grid to user
pactousertotal 0 -1 import from grid to user total
pactogrid 0 -1 export to grid
pactogridtotal 683 -1 export to grid total
```

**Comment by biemond** [comment]
now we need to check some time where we have import and some export.

**Comment by biemond** [comment]
send me a report when you think this is the case

**Comment by MiosDK** [comment]
Report is sent with Import : c0dd83ea-583f-4d20-9969-da1a968ab081

![billede](https://github.com/biemond/solaredge.modbus/assets/170458604/35a176c3-7423-4fb7-96e6-26728cf31273)

**Comment by biemond** [comment]
```
inputPower 0 -1 Input Power
outputPower 0 -1 Output Power
gridOutputPower 0 -1 Grid Output Power
pv1InputPower 1 -1 pv1 Power
pv2InputPower 1 -1 pv2 Power
battDischarge 0 -1 battery Discharge
battCharge 0 -1 battery Charge
pactouserr 54750 -1 import from grid to user
pactousertotal 54617 -1 import from grid to user total
pactogrid 0 -1 export to grid
pactogridtotal 0 -1 export to grid total
```

**Comment by biemond** [comment]
let me check the conversion , maybe something is off.

**Comment by MiosDK** [comment]
> let me check the conversion , maybe something is off.

New report with import from grid : dba984f4-8a11-4d30-8708-6a708a99571b

![Screenshot_20240525_082829_ShinePhone](https://github.com/biemond/solaredge.modbus/assets/170458604/ab3cc35f-419a-478c-a254-8d9e8311078c)

**Comment by biemond** [comment]
```
inputPower 4700 -1 Input Power
outputPower 28303 -1 Output Power
gridOutputCurrent 120 -1 Grid Output Current
gridOutputPower 28456 -1 Grid Output Power
pv1InputPower 2621 -1 pv1 Power
pv2InputPower 2001 -1 pv2 Power
battDischarge 25400 -1 battery Discharge
battCharge 0 -1 battery Charge
battsoc 15 0 battery soc
bmssoc 15 0 bms soc
pactouserr 9580 -1 import from grid to user
pactousertotal 9580 -1 import from grid to user total
pactogrid 0 -1 export to grid
pactogridtotal 0 -1 export to grid total
```

**Comment by biemond** [comment]
so pv, batt discharge and import are ok. 

load = gridOutputPower plus pactouserr or inputPower + battDischarge + pactouserr

**Comment by biemond** [comment]
now we need to have proof of export.

**Comment by MiosDK** [comment]
> now we need to have proof of export.

Now with export to grid : 20adc72b-35b6-4ae8-9dab-18b2c6260385

![Screenshot_20240525_120313_ShinePhone](https://github.com/biemond/solaredge.modbus/assets/170458604/e6d98969-c1bd-4427-bc0b-ad024b7b80fd)

**Comment by biemond** [comment]
I think we got it , will add import and export. 

```
inputPower 32500 -1 Input Power
outputPower 27917 -1 Output Power

gridOutputCurrent 118 -1 Grid Output Current
gridOutputPower 28128 -1 Grid Output Power

pv1InputPower 18272 -1 pv1 Power
pv2InputPower 14262 -1 pv2 Power

battDischarge 0 -1 battery Discharge
battCharge 2200 -1 battery Charge

battsoc 98 0 battery soc
bmssoc 98 0 bms soc
totalhouseload 11100 -1 Total house Load
pactouserr 0 -1 import from grid to user
pactousertotal 0 -1 import from grid to user total
pactogrid 19303 -1 export to grid
pactogridtotal 19255 -1 export to grid total
```

**Comment by biemond** [comment]
can you try this https://homey.app/a/solaredge.modbus/test/

it has now meter import and export as new device fields

**Comment by MiosDK** [comment]
> can you try this https://homey.app/a/solaredge.modbus/test/
> 
> it has now meter import and export as new device fields

YES export is working fantastic, I will try import from grid when the sun is gone and the battery is empty 👍

![Screenshot_20240525_142831_Homey](https://github.com/biemond/solaredge.modbus/assets/170458604/e479e854-9233-4c4c-b385-64d7596c8551)

**Comment by biemond** [comment]
cool, my Danish is not ok but I remember I used Meter export as label.

**Comment by MiosDK** [comment]
> cool, my Danish is not ok but I remember I used Meter export as label.

Yes you have called it Meter Export 👍 could it be an idea to call it the same text as it is in Growatt Shinephone as in Solaredge ? So Meter export is called 'exported to grid' in Solaredge ?

![Screenshot_20240525_150822_Homey](https://github.com/biemond/solaredge.modbus/assets/170458604/f50d558f-8eb3-4f44-84ca-15ae195de00a)

**Comment by biemond** [comment]
next version will have it

**Comment by MiosDK** [comment]
and now with import from Grid : 903383ed-29b0-4111-a377-d30e38c31bcd

![Screenshot_20240526_071329_ShinePhone](https://github.com/biemond/solaredge.modbus/assets/170458604/80bf80f9-eafb-41c5-9fbe-660712834cc6)

**Comment by biemond** [comment]
```
pactouserr 2369 -1 import from grid to user
pactousertotal 2318 -1 import from grid to user total
pactogrid 0 -1 export to grid
pactogridtotal 0 -1 export to grid total
totalhouseload 3500 -1 Total house Load
inputPower 1300 -1 Input Power
```

**Comment by MiosDK** [comment]
Thank you for your help in this matter 👍

**Comment by MiosDK** [comment]
Is this correct or what is the sum of PV1 + PV2 ? I guess that total Day yield is total of pv1 + pv2 but its not showing that ? (See pic) 
![Screenshot_20240602_232231_Homey](https://github.com/biemond/solaredge.modbus/assets/170458604/81aa9b08-dda4-4303-b0ac-6eaba5bd1367)
![Screenshot_20240602_232354_ShinePhone](https://github.com/biemond/solaredge.modbus/assets/170458604/218f90b5-0a9b-4305-9c0b-7ee93af69762)

**Comment by biemond** [comment]
hey, no 

I think total day yield is PV minus your batt charging, when you discharge , PV will be the same  but  total should go up.
